pg_search/src/index/search.rs

```diff
@@ -193,17 +211,14 @@ impl SearchIndex {
         )
     }
 
-    pub fn search_state<W: WriterClient<WriterRequest>>(
-        &self,
+    pub fn search_state<W: WriterClient<WriterRequest> + Send + Sync + 'static>(
+        &mut self,
         writer: &Arc<Mutex<W>>,
         config: &SearchConfig,
-        needs_commit: bool,
     ) -> Result<SearchState, SearchIndexError> {
         // Commit any inserts or deletes that have occurred during this transaction.
-        if needs_commit {
-            writer.lock().request(WriterRequest::Commit {
-                directory: self.directory.clone(),
-            })?
+        if self.is_dirty {
+            self.commit(writer)?;
         }
```

Comment:
we retain the ability to commit the changed index early, we just use our internal state to check if it's necessary instead of it coming externally.